### PR TITLE
Add "Gift options" link to main navigation

### PR DESCRIPTION
**Pull request by GHaberis**
## What does this change?

This changes the main navigation for logged in subscribers and recurring contributors, so they can see a link to "Gift options", which link will take them through to the support.theguardian.com site. By default the link is hidden, we unhide the link in the client once we've verified whether the user is a subscriber/contributor. The "Gift options" link will only appear from the desktop breakpoint up. 

Additional changes:
- The "Gift options" link needs it's own set of tracking properties so we can measure user interactions with this element. I've therefore had to update the code that generates Reader Revenue URLs with the correct tracking properties to do this for the gifting URL. We're only using the instance to be used in the `Header` on web, but I've made the changes to generate the URL for the `SlimHeaderDropdown`, `SideMenu` and `Footer` on web, plus the `AmpHeader` and `AmpFooter`.

Trello: https://trello.com/c/2PwBrI7H/3524-gifting-test-add-a-link-to-subscriptions-from-the-global-masthead

## Does this change need to be reproduced in dotcom-rendering ?

- [x] Yes (please indicate your plans for DCR Implementation)

The link also needs to be present in the navigation on `dotcom-rendering` rendered pages. Reader Revenue links (with all the associated tracking properties) are passed in the data model to `dotcom-rendering`. This ensures the Reader Revenue links on `dotcom-rendering` pages match those on `frontend`. I've therefore had to update the `dotcom-rendering` data model to include a new `gifting` url property.

### desktop breakpoint (980px)
gifting-fe-desktop-980

### leftCol breakpoint (1140px)
gifting-fe-leftCol-1140

### wide breakpoint (1300px)
gifting-fe-wide-1300

## What is the value of this and can you measure success?
This will help subscribers and contributors realise that they could give a subscription to the Guardian as a gift, and get to the support site to do this.

### Does this change break ad-free?

- [x] No

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

**Review by i-hardy — APPROVED** [review]
LGTM 👍

File: common/app/navigation/ReaderRevenueSite.scala [hunk 1]
```diff
@@ -32,4 +32,8 @@ object ReaderRevenueSite extends enumeratum.Enum[ReaderRevenueSite] {
   case object SupportContribute extends ReaderRevenueSite {
     val url: String = s"${Configuration.id.supportUrl}/contribute"
   }
+
+  case object SupportGifting extends ReaderRevenueSite {
+    val url: String = s"${Configuration.id.supportUrl}/subscribe"
```
Comment (johnduffell) [inline common/app/navigation/ReaderRevenueSite.scala:37]: why is this the same as SupportSubscribe?  I guess we don't have a gift landing page?
https://support.theguardian.com/subscribe

File: static/src/inline-svgs/icon/gifting.svg [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<svg width="16" height="18" viewBox="0 0 16 18" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (johnduffell) [inline static/src/inline-svgs/icon/gifting.svg:1]: has this been run through svgo? https://jakearchibald.github.io/svgomg/

**Review by johnduffell — APPROVED** [review]
looks good, just a couple of questions 👍

File: common/app/navigation/ReaderRevenueSite.scala (hunk 1 above)
Comment (GHaberis) [reply common/app/navigation/ReaderRevenueSite.scala:37]: Exactly that

File: static/src/inline-svgs/icon/gifting.svg (hunk 2 above)
Comment (johnduffell) [reply static/src/inline-svgs/icon/gifting.svg:1]:
```xml
<svg width="16" height="18" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M4.5 4.3c1 0 1.8-.7 2.5-2L4 0h-.7l-.6 2.4.4.3c.7 1.6.5 1.6 1.4 1.6zm7 0c-1 0-1.8-.7-2.5-2L11.8 0h.8l.6 2.4.1.3c-.7 1.6-1 1.6-1.8 1.6z" fill="#fff"/><path d="M6.5 6.6H1V3.4l.5-.5h5v3.7zm0 2.7v7.6H2l-.5-.5V9.3h5zM15 6.6H9.5V2.9h5l.5.5v3.2zm-.5 2.7v7.1l-.5.5H9.5V9.3h5z" fill="#fff" stroke="#fff"/></svg>
```
image
image

**Pull request merged**